bus-mapping/src/evm/opcodes/callop.rs

```diff
@@ -246,7 +266,15 @@ impl<const N_ARGS: usize> Opcode for CallOpcode<N_ARGS> {
                         CallContextField::ReturnDataLength,
                         call.return_data_length.into(),
                     ),
-                    (CallContextField::Value, call.value),
+                    (
+                        CallContextField::Value,
+                        // Should set to value of current call for DELEGATECALL.
+                        if call.kind == CallKind::DelegateCall {
+                            current_call.value
+                        } else {
+                            call.value
+                        },
+                    ),
```

Comment:
Hi @han0110 , as your comment https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/973#issuecomment-1363563575, previously I think this fix should have the same effect with https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/975 (and no special condition for `has_value`). What do you think? Thanks.

Reply:
This fixes the case when there is a single `DELEGATECALL`, but when there recursive `DELEGATECALL` to self, the logic here will only propagate it 1 layer (so for deeper layers the value will be `0` which is not correct).

Reply:
I understand, thanks for explanation. Please let me update this PR and add the fixes of https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/975. Thanks a lot.

Reply:
Thank you!

Reply:
Merged https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/975 into this PR. And updated `has_value` in circuit code as:
```
let has_value = select::expr(is_delegatecall.expr(), 0.expr(), 1.expr() - value_is_zero.expr());
```
And in function `assign_exec_step`:
```
let has_value = !value.is_zero() && !is_delegatecall;
```